Index: slstatus/config.def.h
===================================================================
--- slstatus.orig/config.def.h	2023-11-19 18:40:43.053731656 +0500
+++ slstatus/config.def.h	2023-11-26 13:09:26.860313413 +0500
@@ -65,5 +65,7 @@
  */
 static const struct arg args[] = {
 	/* function format          argument */
-	{ datetime, "%s",           "%F %T" },
+	{ keymap,   " %s |",         NULL },
+	{ vol_perc, " VOL %2s%% |",  "/dev/mixer" },
+	{ datetime, " %s",           "%F %T"  },
 };
