Index: slstatus/components/keymap.c
===================================================================
--- slstatus.orig/components/keymap.c	2023-11-29 22:36:50.247222032 +0500
+++ slstatus/components/keymap.c	2023-11-29 22:41:54.026901859 +0500
@@ -40,7 +40,11 @@
 		layout = tok;
 		grp++;
 	}
-
+	char *up = strtok(layout, ":");
+	while (*up) {
+	  *up = toupper((unsigned char) *up);
+	  up++;
+	}
 	return layout;
 }
 
