Index: st/config.def.h
===================================================================
--- st.orig/config.def.h	2023-11-26 12:48:30.720699052 +0500
+++ st/config.def.h	2023-11-26 12:50:39.727464024 +0500
@@ -5,7 +5,7 @@
  *
  * font: see http://freedesktop.org/software/fontconfig/fontconfig-user.html
  */
-static char *font = "Liberation Mono:pixelsize=12:antialias=true:autohint=true";
+static char *font = "JetBrainsMono NFM:size=10:antialias=true:autohint=true";
 static int borderpx = 2;
 
 /*
