Index: dwm/config.def.h
===================================================================
--- dwm.orig/config.def.h	2023-11-18 20:32:16.123717080 +0100
+++ dwm/config.def.h	2023-11-18 20:32:41.073717611 +0100
@@ -6,8 +6,8 @@
 static const unsigned int snap      = 32;       /* snap pixel */
 static const int showbar            = 1;        /* 0 means no bar */
 static const int topbar             = 1;        /* 0 means bottom bar */
-static const char *fonts[]          = { "monospace:size=10" };
-static const char dmenufont[]       = "monospace:size=10";
+static const char *fonts[]          = { "JetBrains Mono:size=10" };
+static const char dmenufont[]       = "JetBrains Mono:size=10";
 static const char col_gray1[]       = "#222222";
 static const char col_gray2[]       = "#444444";
 static const char col_gray3[]       = "#bbbbbb";
