Index: dwm/config.def.h
===================================================================
--- dwm.orig/config.def.h	2023-11-26 13:06:46.860462928 +0500
+++ dwm/config.def.h	2023-11-26 13:07:36.333755786 +0500
@@ -7,15 +7,17 @@
 static const int topbar             = 1;        /* 0 means bottom bar */
 static const char *fonts[]          = { "JetBrains Mono:size=10" };
 static const char dmenufont[]       = "JetBrains Mono:size=10";
-static const char col_gray1[]       = "#222222";
-static const char col_gray2[]       = "#444444";
-static const char col_gray3[]       = "#bbbbbb";
-static const char col_gray4[]       = "#eeeeee";
-static const char col_cyan[]        = "#005577";
+/* Set colors for dwm */
+static const char norm_fg[]         = "#0abdc6";
+static const char norm_bg[]         = "#091833";
+static const char norm_border[]     = "#091833";
+static const char sel_fg[]          = "#ea00d9";
+static const char sel_bg[]          = "#321959";
+static const char sel_border[]      = "#ea00d9";
 static const char *colors[][3]      = {
-	/*               fg         bg         border   */
-	[SchemeNorm] = { col_gray3, col_gray1, col_gray2 },
-	[SchemeSel]  = { col_gray4, col_cyan,  col_cyan  },
+    /*               fg       bg       border   */
+    [SchemeNorm] = { norm_fg, norm_bg, norm_border },
+    [SchemeSel]  = { sel_fg,  sel_bg,  sel_border  },
 };
 
 /* tagging */
@@ -57,7 +59,7 @@
 
 /* commands */
 static char dmenumon[2] = "0"; /* component of dmenucmd, manipulated in spawn() */
-static const char *dmenucmd[] = { "dmenu_run", "-m", dmenumon, "-fn", dmenufont, "-nb", col_gray1, "-nf", col_gray3, "-sb", col_cyan, "-sf", col_gray4, NULL };
+static const char *dmenucmd[] = { "dmenu_run", "-m", dmenumon, "-fn",dmenufont, "-nb", norm_bg, "-nf", norm_fg, "-sb", sel_bg, "-sf", sel_fg, NULL };
 static const char *termcmd[]  = { "st", NULL };
 
 static const Key keys[] = {
